app/src/main/java/pl/droidevs/books/library/LibraryActivity.java

```diff
@@ -1,15 +1,77 @@
 package pl.droidevs.books.library;
 
+import android.arch.lifecycle.ViewModelProvider;
+import android.arch.lifecycle.ViewModelProviders;
 import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
 import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.ProgressBar;
 
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+import dagger.android.AndroidInjection;
 import pl.droidevs.books.R;
 
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+
 public class LibraryActivity extends AppCompatActivity {
 
+    @BindView(R.id.layout_books)
+    RecyclerView recyclerView;
+
+    @BindView(R.id.progress_books)
+    ProgressBar progressBar;
+
+    @BindView(R.id.button_add_book)
+    FloatingActionButton floatingActionButton;
+
+    @Inject
+    ViewModelProvider.Factory viewModelFactory;
+
+    private Unbinder unbinder;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_library);
+
+        AndroidInjection.inject(this);
+        this.unbinder = ButterKnife.bind(this);
+
+        final LibraryAdapter adapter = new LibraryAdapter();
+        adapter.setItemClickListener(bookId -> {
+            // TODO: Start details activity and pass the book id
+        });
+        recyclerView.setAdapter(adapter);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+
+        floatingActionButton.setOnClickListener(view -> {
+            // TODO: Start Add book activity
+        });
+
+        final LibraryViewModel libraryViewModel = ViewModelProviders.of(this, viewModelFactory).get(LibraryViewModel.class);
+        libraryViewModel.getBooks().observe(this, books -> {
+            progressBar.setVisibility(GONE);
+            if (books != null)
```

Comment:
Ja wiem, że się czepiam, ale if bez {} bardzo mnie kłuje w oczy. Oczywiście możemy ustalić, że tak może być i zostawiamy - wtedy będę musiała się nauczyć z tym żyć :D

Reply:
Dla mnie nie ma problemu, w pracy też jestem zmuszony używać klamr, więc jestem przyzwyczajony :wink: Chociaż sam zgadzam się z Uncle Bobem: One of my favorite things to do is getting rid of useless braces :smile: https://softwareengineering.stackexchange.com/questions/320247/has-can-anyone-challenge-uncle-bob-on-his-love-of-removing-useless-braces